include/cuco/detail/static_multimap.inl

```diff
@@ -0,0 +1,1153 @@
+
+namespace cuco {
+
+/**---------------------------------------------------------------------------*
+ * @brief Enumeration of the possible results of attempting to insert into
+ *a hash bucket
+ *---------------------------------------------------------------------------**/
+enum class insert_result {
+  CONTINUE,  ///< Insert did not succeed, continue trying to insert
+  SUCCESS,   ///< New pair inserted successfully
+  DUPLICATE  ///< Insert did not succeed, key is already present
+};
+
+template <typename Key,
+          typename Value,
+          class ProbeSequence,
+          cuda::thread_scope Scope,
+          typename Allocator>
+static_multimap<Key, Value, ProbeSequence, Scope, Allocator>::static_multimap(
+  std::size_t capacity, Key empty_key_sentinel, Value empty_value_sentinel, Allocator const& alloc)
+  : empty_key_sentinel_{empty_key_sentinel},
+    empty_value_sentinel_{empty_value_sentinel},
+    slot_allocator_{alloc}
+{
+  if constexpr (is_vector_load()) {
+    capacity_ = cuco::detail::get_valid_capacity<cg_size() * 2>(capacity);
+  } else {
+    capacity_ = cuco::detail::get_valid_capacity<cg_size()>(capacity);
+  }
+
+  slots_ = std::allocator_traits<slot_allocator_type>::allocate(slot_allocator_, get_capacity());
+
+  auto constexpr block_size = 256;
+  auto constexpr stride     = 4;
+  auto const grid_size      = (get_capacity() + stride * block_size - 1) / (stride * block_size);
+  detail::initialize<atomic_key_type, atomic_mapped_type>
+    <<<grid_size, block_size>>>(slots_, empty_key_sentinel, empty_value_sentinel, get_capacity());
+}
+
+template <typename Key,
+          typename Value,
+          class ProbeSequence,
+          cuda::thread_scope Scope,
+          typename Allocator>
+static_multimap<Key, Value, ProbeSequence, Scope, Allocator>::~static_multimap()
+{
+  std::allocator_traits<slot_allocator_type>::deallocate(slot_allocator_, slots_, capacity_);
+}
+
+template <typename Key,
+          typename Value,
+          class ProbeSequence,
+          cuda::thread_scope Scope,
+          typename Allocator>
+template <typename InputIt, typename KeyEqual>
+void static_multimap<Key, Value, ProbeSequence, Scope, Allocator>::insert(InputIt first,
+                                                                          InputIt last,
+                                                                          cudaStream_t stream,
+                                                                          KeyEqual key_equal)
+{
+  auto num_keys         = std::distance(first, last);
+  auto const block_size = 128;
+  auto const stride     = 1;
+  auto const grid_size  = (cg_size() * num_keys + stride * block_size - 1) / (stride * block_size);
+  auto view             = get_device_mutable_view();
+
+  detail::insert<block_size, cg_size(), is_vector_load()>
+    <<<grid_size, block_size, 0, stream>>>(first, first + num_keys, view, key_equal);
+  CUCO_CUDA_TRY(cudaDeviceSynchronize());
```

Comment:
Good point. This logic is taken from `static_map` also and in `static_multimap` we added stream as an argument so we can definitely use stream synchronization instead. I will remove the synchronization here since it's indeed unnecessary but for other host-bulk functions like `count` or `retrieve`, I guess we still need to have stream sync to make sure the global counter is properly updated? @jrhemstad please confirm.